server/src/main/java/org/apache/iotdb/db/query/reader/series/SeriesReader.java

```diff
@@ -381,19 +383,19 @@ boolean hasNextPage() throws IOException {
       /*
        * first chunk metadata is already unpacked, consume cached pages
        */
-      if (!cachedPageReaders.isEmpty()) {
-        firstPageReader = cachedPageReaders.poll();
+      initFirstPageReader();
+      if (firstPageReader != null) {
         long endpointTime = orderUtils.getOverlapCheckTime(firstPageReader.getStatistics());
         unpackAllOverlappedTsFilesToTimeSeriesMetadata(endpointTime);
         unpackAllOverlappedTimeSeriesMetadataToCachedChunkMetadata(endpointTime, false);
         unpackAllOverlappedChunkMetadataToCachedPageReaders(endpointTime, false);
```

Comment:
Since you modified the field name of `cachedPageReaders`, maybe the name of this method should be updated as well.

Reply:
Remaining this name is still better I think.